lib/Settings/SettingsController.php

```diff
@@ -27,7 +27,7 @@ function get($key, $default = null) {
       $default = $this->getDefaultValue($key_parts);
     }
     foreach ($key_parts as $key_part) {
-      if (array_key_exists($key_part, $setting)) {
+      if (is_array($setting) && array_key_exists($key_part, $setting)) {
```

Comment:
We have [an integration test](https://github.com/mailpoet/mailpoet/blob/master/tests/integration/Settings/SettingsControllerTest.php) for this controller. Please add a test for this issue so that we don't loose the functionality with some refactor.

Reply:
Nice suggestion, while working on the test I realized that the problem in fact originates from the `fetch()` method which, when called with a non-existent nested key name, produces a setting value that `get()` cannot access properly.